src/components/ContentNode/Caption.vue

```diff
@@ -53,4 +58,8 @@ export default {
 /deep/ p {
   display: inline-block;
 }
+
+caption {
```

Comment:
I think we only want a top margin to be applied if it is preceded by some other content, right?

It might be worth looking at the way this is/was done for the figcaption/figure spacing in [`Figure.vue`][1]. It sounds like we would want to do something similar for the `caption` element and the contents within a table.

[1]: https://github.com/marinaaisa/swift-docc-render/blob/f81e35c7391f25cce99ff7385dc446ea1804aa76/src/components/ContentNode/Figure.vue#L28

Reply:
Because we are adding the caption inside the table, I couldn't apply the margin to its siblings, it won't work in a display table, so I applied it to the component itself checking if there are siblings, using the `has` pseudo-class. It's widely supported now (only Firefox is missing) https://caniuse.com/css-has https://github.com/apple/swift-docc-render/pull/692/commits/d814cd80923aba1bde7e919e7817c6c9d70b1415

Can we use `has`?

Reply:
I started reviewing this again, but I found out that the `<caption>` element has to be the first element of the `<table>`, regardless of whether or not it is displayed on the top or the bottom. 🤦‍♂️🥲

I was going to make some suggestions to this PR, but it turned out to just create a new PR. Hopefully we finally have it right now! https://github.com/apple/swift-docc-render/pull/706

I also ran into issues you probably did where our normal block-level spacing helpers don't work due to specific issues with certain elements, but I was able to address them without `:has()`.